### PR TITLE
Loss function

### PR DESCRIPTION
**Pull request by OleBialas**
Replaced the hard coded estimation of r and mse with a loss function. The loss function defaults to one_minus_correlation but can be set to anything that takes y and y_pred as input and retruns a scalar who's value can be minimized to optimize regularization. Also added some minor changes to improve code readability and consistency

### CODE REVIEW

File: mtrf/matrices.py
Comment (StellarBlocks-Assistant) [inline mtrf/matrices.py:1]: Still believe move the if..else branch outside will make the code more efficient (to avoid repeatedly checking the same value). Calculating the covariance matrix is one of the core functionality of this package, we should keep it as efficient as we can.

File: mtrf/matrices.py [hunk 1]
```diff
@@ -132,24 +132,19 @@ def covariance_matrices(x, y, lags, zeropad=True, preload=True):
     x, y, _ = _check_data(x, y)
     if zeropad is False:
         y = truncate(y, lags[0], lags[-1])
-
-    x_lag_size = x[0].shape[-1] * len(lags) + 1
```
Comment (StellarBlocks-Assistant) [inline mtrf/matrices.py:136]: Still believe move the if..else branch outside will make the code more efficient (to avoid repeatedly checking the same value). Calculating the covariance matrix is one of the core functionality of this package, we should keep it as efficient as we can.

Comment (powerfulbean) [inline mtrf/matrices.py:136]: Still believe move the if..else branch outside will make the code more efficient (to avoid repeatedly checking the same value). Calculating the covariance matrix is one of the core functionality of this package, we should keep it as efficient as we can.

**Comment by powerfulbean** [comment]
I think here we may need to distinguish the function for selecting best hyperparameter (the loss function here we want), and the real loss function of our linear regression solver, which would be mean square error. So, I don't think it is necessary to not letting predict return both mse and pearson'r, since they are the necessary metrics to evaluate a linear regression model.
